osu.Game/OsuGameBase.cs

```diff
@@ -421,6 +421,19 @@ private void load(ReadableKeyCombinationProvider keyCombinationProvider, Framewo
 
             Ruleset.BindValueChanged(onRulesetChanged);
             Beatmap.BindValueChanged(onBeatmapChanged);
+
+            // make config aware of how to lookup skins for on-screen display purposes.
+            // if this becomes a more common thing, tracked settings should be reconsidered to allow local DI.
+            LocalConfig.LookupSkinNameFunc = id => SkinManager.Query(s => s.ID == id)?.ToString() ?? "Unknown";
+            LocalConfig.LookupKeyBindingsFunc = l =>
+            {
+                var combinations = KeyBindingStore.GetReadableKeyCombinationsFor(l);
+
+                if (combinations.Count == 0)
+                    return ToastStrings.NoKeyBound;
+
+                return string.Join(" / ", combinations);
+            };
```

Comment:
dear lord this exists in master already??? how did this happen.

looking away from whatever this is